plugins/explore/src/components/GroupsExplorerContent/GroupsDiagram.tsx

```diff
@@ -91,19 +109,86 @@ function RenderNode(props: DependencyGraphTypes.RenderNodeProps<any>) {
         })}
       >
         <text
-          x={90}
-          y={45}
+          x={NODE_WIDTH / 2}
+          y={NODE_HEIGHT / 2 + NODE_MIDDLE_ALIGNMENT_SHIFT}
           textAnchor="middle"
           alignmentBaseline="baseline"
-          style={{ fontWeight: 'bold' }}
+          style={{ fontWeight: 'bold', fontSize: TEXT_FONT_SIZE }}
         >
-          {props.node.name}
+          {objs.map(function (object: any, i: number) {
+            return (
+              <tspan
+                key={i}
+                y={NODE_HEIGHT / 2 + NODE_MIDDLE_ALIGNMENT_SHIFT}
+                x="90"
+                textAnchor="middle"
+                dy={object.dy}
+              >
+                {object.text}
+              </tspan>
+            );
+          })}
         </text>
       </Link>
     </g>
   );
 }
 
+/**
+ * Join chunks based on max words per line and add dy position shifting
+ * @param chunkedArray
+ * @returns
+ */
+function prepareForDisplay(chunkedArray: any): any {
+  return chunkedArray.map((val: Array<string>, pos: number) => {
+    const text = val.join(' ');
+    // If array length == 1, dy should be 0 since there is no need to divide the node into blocks, we just need to centralize it
+    const dy = chunkedArray.length === 1 ? 0 : getDy(pos) - 30;
+    return { dy: dy, text: text };
+  });
+}
+
+/**
+ * text svg dy shifting based on array pos and in the blocks inside the node
+ * @param i text position on the array
+ */
+function getDy(i: number) {
+  const blocksSize = NODE_HEIGHT / NODE_MAX_LINES;
+  const position = blocksSize * i;
+  return position;
+}
+
+/**
+ * Create Chunked name based on NODE_MAX_LINES and NODE_MAX_WORDS_PER_ROW
+ * @param name from props.node.name
+ * @returns
+ */
+function splitNameInChunks(name: string) {
+  const array = name.split(' ');
+  const formated = array
+    .slice(0, NODE_MAX_LINES * NODE_MAX_WORDS_PER_ROW)
+    .map((it, pos) =>
+      pos + 1 === NODE_MAX_LINES * NODE_MAX_WORDS_PER_ROW ? `${it}...` : it,
+    );
+  const chunked = chunkArray(formated, NODE_MAX_WORDS_PER_ROW);
+  return chunked;
+}
+
+/**
+ * Returns an array with arrays of the given size.
+ *
+ * @param arr {Array} Array to split
+ * @param size {Integer} Size of each group
+ */
+function chunkArray(arr: Array<any>, size: number) {
```

Comment:
we already import lodash and use it a lot, you can leverage https://lodash.com/docs/4.17.15#chunk

Reply:
removed, since css is doing all the tricky